### PR TITLE
Update example lifecycle methods

### PR DESCRIPTION
**Pull request by gretzky**
Some examples and docs are using unsafe lifecycle methods (`componentWillReceiveProps` and `componentWillMount`) despite being updated to latest React. I've replaced these methods with their correct counterparts and updated the docs to reflect them as well.

Fixes #3218

### CODE REVIEW

**Comment by gretzky** [comment]
@timdorr changed, do you see the need to add `getDerivedStateFromProps` as well?

File: docs/api/applyMiddleware.md [hunk 1]
```diff
@@ -172,9 +172,9 @@ class SandwichShop extends Component {
     this.props.dispatch(makeASandwichWithSecretSauce(this.props.forPerson))
   }
 
-  componentWillReceiveProps(nextProps) {
-    if (nextProps.forPerson !== this.props.forPerson) {
-      this.props.dispatch(makeASandwichWithSecretSauce(nextProps.forPerson))
+  componentDidUpdate(prevProps) {
+    if (prevProps.forPerson !== this.props.forPerson) {
+      this.props.dispatch(makeASandwichWithSecretSauce(prevProps.forPerson))
```
Comment (timdorr) [inline docs/api/applyMiddleware.md]: This should be `this.props.forPerson`.

File: docs/recipes/ReducingBoilerplate.md [hunk 2]
```diff
@@ -221,9 +221,9 @@ class Posts extends Component {
     this.loadData(this.props.userId)
   }
 
-  componentWillReceiveProps(nextProps) {
-    if (nextProps.userId !== this.props.userId) {
-      this.loadData(nextProps.userId)
+  componentDidUpdate(prevProps) {
+    if (prevProps.userId !== this.props.userId) {
+      this.loadData(prevProps.userId)
```
Comment (timdorr) [inline docs/recipes/ReducingBoilerplate.md]: This should be `this.props.userId`

File: docs/recipes/ReducingBoilerplate.md [hunk 3]
```diff
@@ -306,9 +306,9 @@ class Posts extends Component {
     this.props.dispatch(loadPosts(this.props.userId))
   }
 
-  componentWillReceiveProps(nextProps) {
-    if (nextProps.userId !== this.props.userId) {
-      this.props.dispatch(loadPosts(nextProps.userId))
+  componentDidUpdate(prevProps) {
+    if (prevProps.userId !== this.props.userId) {
+      this.props.dispatch(loadPosts(prevProps.userId))
```
Comment (timdorr) [inline docs/recipes/ReducingBoilerplate.md]: This should be `this.props.userId`

File: examples/async/src/containers/App.js [hunk 4]
```diff
@@ -19,9 +19,9 @@ class App extends Component {
     dispatch(fetchPostsIfNeeded(selectedSubreddit))
   }
 
-  componentWillReceiveProps(nextProps) {
-    if (nextProps.selectedSubreddit !== this.props.selectedSubreddit) {
-      const { dispatch, selectedSubreddit } = nextProps
+  componentDidUpdate(prevProps) {
+    if (prevProps.selectedSubreddit !== this.props.selectedSubreddit) {
+      const { dispatch, selectedSubreddit } = prevProps
```
Comment (timdorr) [inline examples/async/src/containers/App.js]: This should be using `this.props` instead.

File: examples/real-world/src/components/Explore.js [hunk 5]
```diff
@@ -11,9 +11,9 @@ export default class Explore extends Component {
     onChange: PropTypes.func.isRequired
   }
 
-  componentWillReceiveProps(nextProps) {
-    if (nextProps.value !== this.props.value) {
-      this.setInputValue(nextProps.value)
+  componentDidUpdate(prevProps) {
+    if (prevProps.value !== this.props.value) {
+      this.setInputValue(prevProps.value)
```
Comment (timdorr) [inline examples/real-world/src/components/Explore.js]: This should be `this.props.value`

File: examples/real-world/src/containers/RepoPage.js [hunk 6]
```diff
@@ -27,13 +27,13 @@ class RepoPage extends Component {
...
+  componentDidMount() {
     loadData(this.props)
   }
 
-  componentWillReceiveProps(nextProps) {
-    if (nextProps.fullName !== this.props.fullName) {
-      loadData(nextProps)
+  componentDidUpdate(prevProps) {
+    if (prevProps.fullName !== this.props.fullName) {
+      loadData(prevProps)
```
Comment (timdorr) [inline examples/real-world/src/containers/RepoPage.js]: This should be `this.props`.

File: examples/real-world/src/containers/UserPage.js [hunk 7]
```diff
@@ -26,13 +26,13 @@ class UserPage extends Component {
...
+  componentDidMount() {
     loadData(this.props)
   }
 
-  componentWillReceiveProps(nextProps) {
-    if (nextProps.login !== this.props.login) {
-      loadData(nextProps)
+  componentDidUpdate(prevProps) {
+    if (prevProps.login !== this.props.login) {
+      loadData(prevProps)
```
Comment (timdorr) [inline examples/real-world/src/containers/UserPage.js]: This should be `this.props`

**Comment by gretzky** [comment]
My fault, I didn't commit those. I'll run through your comments and make sure I got them all.

**Comment by timdorr** [comment]
Thanks!

**Pull request merged**